src/connector.js

```diff
@@ -896,11 +901,22 @@ function updateQuerySummaryState( newState ) {
 		querySummaryElement.textContent = "";
 		if ( querySummaryState.total > 0 ) {
 			let numberOfResults = querySummaryState.total.toLocaleString( params.lang );
-
-			querySummaryElement.innerHTML = ( ( querySummaryState.query !== "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
-				.replace( '%[numberOfResults]', numberOfResults )
-				.replace( '%[query]', DOMPurify.sanitize( querySummaryState.query ) )
-				.replace( '%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString( params.lang ) );
+			// Create the <h2> element
+			const hTwoAnchor = document.createElement("h2");
+			// Generate the text content
+			const querySummaryText = ((querySummaryState.query !== "" && !params.isAdvancedSearch) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML)
+				.replace('%[numberOfResults]', numberOfResults)
+				.replace('%[query]', querySummaryState.query)
+				.replace('%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString(params.lang));
+
+			const dirty = querySummaryText
+			const clean = DOMPurify.sanitize(dirty, {
+				ALLOWED_TAGS: [""],
+				FORBID_ATTR: ["style"],
+			});
+			hTwoAnchor.textContent = clean;
```

Comment:
Sanitizing the query text is no longer needed as it's not output as HTML anymore. Although, the template will need to be updated to remove the H2 (in `querySummaryTemplateHTML` and `noQuerySummaryTemplateHTML`).

```suggestion
			hTwoAnchor.textContent = querySummaryText;
```

Reply:
@Garneauma  Resolved, please see my changes

Reply:
Resolved